first_validation.py

```diff
@@ -54,7 +53,16 @@ def check_hashes(hash_fname):
         # Calculate actual hash for given filename.
         # Check actual hash against expected hash
         # Return False if any of the hashes do not match.
-    return False
+    hash_text = hash_pth.read_text().splitlines()
+    print(hash_text)
+    for line in hash_text:
+        hash, file = line.split()
+        filepath = data_dir / file
+        if hash != hash_for_fname(filepath):
+            return False
```

Comment:
It's neater to put the comments (above) with the matching code - or remove the comments.